analysis/task_analysis_surface/group_level_analysis_surf.ipynb

```diff
@@ -0,0 +1,994 @@
+    "    iqr_cutoffs_df = pd.read_csv(f'../../../derivatives/mriqc_summaries/iqr_cutoffs_ses-{ses}_rec-unco.tsv', low_memory=False, sep='\\t')\n",
+    "    \n",
+    "    #read in tsv file with mriqc summary for current task, ses, group combi\n",
+    "    mriqc_summary = pd.read_csv(f'../../../derivatives/mriqc_summaries/group-{group}_ses-{ses}_task-{task}_rec-unco.tsv', low_memory=False, sep='\\t')\n",
+    "    \n",
+    "    \n",
+    "    #possible outlier types that use IQR-based cutoff: tsnr, snr, gsr_x, gsr_y\n",
+    "    #exclude runs based on these if in desired_outliers list\n",
+    "    for outlier_type in ['tsnr', 'snr', 'gsr_x', 'gsr_y']:\n",
+    "        if outlier_type in desired_outliers.keys():\n",
+    "            #get upper and lower cutoffs from iqr summaries df\n",
+    "            lower_bound = iqr_cutoffs_df[outlier_type].iloc[0]\n",
+    "            upper_bound = iqr_cutoffs_df[outlier_type].iloc[1]\n",
+    "            \n",
+    "            #find subs_runs combinations outside of cutoffs \n",
+    "            above_subs_runs = mriqc_summary['subs_runs'][mriqc_summary[outlier_type]>=upper_bound].tolist()\n",
+    "            below_subs_runs = mriqc_summary['subs_runs'][mriqc_summary[outlier_type]<=lower_bound].tolist()\n",
+    "            subs_runs = above_subs_runs+below_subs_runs\n",
+    "            excluded_runs+=subs_runs\n",
+    "            \n",
+    "    if 'fd_mean' in desired_outliers.keys():\n",
+    "        above_fd_mean_runs = mriqc_summary['subs_runs'][mriqc_summary['fd_mean']>desired_outliers['fd_mean']].tolist()\n",
+    "        if 'fd_perc' in desired_outliers:\n",
+    "            below_fd_perc_runs = set(mriqc_summary['subs_runs'][mriqc_summary['fd_perc']<desired_outliers['fd_perc']].tolist())\n",
+    "            above_fd_mean_runs = set(above_fd_mean_runs)\n",
+    "            above_fd_mean_runs = list(above_fd_mean_runs.difference(below_fd_perc_runs))\n",
+    "        excluded_runs+=above_fd_mean_runs\n",
+    "        \n",
+    "    #remove duplicates\n",
+    "    excluded_runs=list(set(excluded_runs))\n",
+    "    \n",
+    "    #turn into dict\n",
+    "    for sub_run in excluded_runs:\n",
+    "        sub = sub_run.split('_')[0]\n",
+    "        run = sub_run.split('_')[1]\n",
+    "        if sub in excluded_runs_dict.keys():\n",
+    "            excluded_runs_dict[sub].append(run)\n",
+    "        else:\n",
+    "            excluded_runs_dict[sub]=[run]\n",
+    "    \n",
+    "    return excluded_runs_dict\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2b3ee165",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def rm_CUD_baseline(effect_size_maps, all_subs):\n",
+    "    \n",
+    "    #subs to be excluded (only MM)\n",
+    "    excluded_subs = ['MM014','MM188','MM197','MM217','MM228','MM239','MM241']\n",
```

Comment:
a comment here would be useful why these were excluded.

Reply:
added! and did same for volumetric group analysis